src/js/libcs/Operators.js

```diff
@@ -3607,30 +3691,143 @@ evaluator.allpoints$0 = function(args, modifs) {
     };
 };
 
+evaluator.allpoints$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "P", csgeo.points);
+};
+
+evaluator.allpoints$1 = function(args, modifs) {
+    return eval_helper.allhelper1arg(args, "P");
+};
+
 evaluator.allmasses$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "Mass", masses);
+};
+
+// collects only points
+evaluator.allmasses$1 = function(args, modifs) {
     var erg = [];
-    for (var i = 0; i < masses.length; i++) {
-        erg[i] = {
-            ctype: "geo",
-            value: masses[i],
-            type: "P"
-        };
+    var pts = eval_helper.allhelper1arg(args, "P");
+    var pp;
+    for (var i = 0; i < pts.value.length; i++) {
+        pp = pts.value[i];
+        if (pp.value.behavior && pp.value.behavior.type === "Mass") erg.push(pp);
     }
     return {
         ctype: "list",
         value: erg
     };
+
+};
+
+evaluator.allsprings$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "Spring", springs);
+};
+
+
+// collects only segments
+evaluator.allsprings$1 = function(args, modifs) {
+    var erg = [];
+    var pts = eval_helper.allhelper1arg(args, "S", "Segment");
+    var sp;
+    for (var i = 0; i < pts.value.length; i++) {
+        sp = pts.value[i];
+        if (sp.value.behavior && sp.value.behavior.type === "Spring") erg.push(sp);
+    }
+    return {
+        ctype: "list",
+        value: erg
+    };
+
 };
 
 evaluator.alllines$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "L", csgeo.lines);
+};
+
+evaluator.alllines$1 = function(args, modifs) {
+    return eval_helper.allhelper1arg(args, "L");
+};
+
+
+evaluator.allsegments$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "S", csgeo.lines, "Segment");
+};
+
+
+evaluator.allsegments$1 = function(args, modifs) {
+    return eval_helper.allhelper1arg(args, "S", "Segment");
+};
+
+
+evaluator.allconics$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "L", csgeo.conics);
+};
+
+evaluator.allconics$1 = function(args, modifs) {
+    return eval_helper.allhelper1arg(args, "C");
+};
+
+evaluator.allcircles$0 = function(args, modifs) {
+    return eval_helper.allhelper0arg(args, "C", csgeo.conics, "Circle");
+};
+
+evaluator.allcircles$1 = function(args, modifs) {
+    return eval_helper.allhelper1arg(args, "C", "Circle");
+};
+
+evaluator.allelements$0 = function(args, modifs) {
     var erg = [];
-    for (var i = 0; i < csgeo.lines.length; i++) {
-        erg[i] = {
+
+    var points = evaluator.allpoints$0(null, null);
+    var masses = evaluator.allmasses$0(null, null);
+    var springs = evaluator.allsprings$0(null, null);
+
+    // no segments or circles since we consider them as lines 
+    var lines = evaluator.alllines$0(null, null);
+    var conics = evaluator.allsegments$0(null, null);
```

Comment:
Why so complicated? `csgeo.gslp` has all the elements we want. Including texts and so on.